src/model/guild/automod.rs

```diff
@@ -0,0 +1,859 @@
+        }
+
+        deserializer.deserialize_any(Visitor)
+    }
+}
+
+impl Serialize for Rule {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        #[derive(Serialize)]
+        #[serde(rename = "TriggerMetadata")]
+        struct TriggerMetadataKeyword<'a> {
+            keyword_filter: &'a Vec<String>,
+        }
+
+        #[derive(Serialize)]
+        #[serde(rename = "TriggerMetadata")]
+        struct TriggerMetadataPresent<'a> {
+            presets: &'a Vec<KeywordPresentType>,
+        }
+
+        #[derive(Serialize)]
+        #[serde(rename = "TriggerMetadata")]
+        struct TriggerMetadataEmpty {}
+
+        let mut s = serializer.serialize_struct("Rule", 11)?;
+        s.serialize_field("id", &self.id)?;
+        s.serialize_field("guild_id", &self.guild_id)?;
+        s.serialize_field("name", &self.name)?;
+        s.serialize_field("creator_id", &self.creator_id)?;
+        s.serialize_field("event_type", &self.event_type)?;
+        match &self.trigger {
+            Trigger::Keyword(keyword_filter) => {
+                s.serialize_field("trigger_type", &TriggerType::Keyword)?;
+                s.serialize_field("trigger_metadata", &TriggerMetadataKeyword {
+                    keyword_filter,
+                })?;
+            },
+            Trigger::KeywordPresent(presets) => {
+                s.serialize_field("trigger_type", &TriggerType::KeywordPresent)?;
+                s.serialize_field("trigger_metadata", &TriggerMetadataPresent {
+                    presets,
+                })?;
+            },
+            trigger => {
+                s.serialize_field("trigger_type", &trigger.kind())?;
+                s.serialize_field("trigger_metadata", &TriggerMetadataEmpty {})?;
+            },
+        }
+        s.serialize_field("actions", &self.actions)?;
+        s.serialize_field("enabled", &self.enabled)?;
+        s.serialize_field("exempt_roles", &self.exempt_roles)?;
+        s.serialize_field("exempt_channels", &self.exempt_channels)?;
+        s.end()
+    }
+}
+
+/// Indicates in what event context a rule should be checked.
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Deserialize, Serialize)]
+#[serde(from = "u8", into = "u8")]
+#[non_exhaustive]
+pub enum EventType {
+    MessageSend,
+    Unknown(u8),
+}
+
+impl From<u8> for EventType {
+    fn from(value: u8) -> Self {
+        match value {
+            1 => Self::MessageSend,
+            _ => Self::Unknown(value),
+        }
+    }
+}
+
+impl From<EventType> for u8 {
+    fn from(value: EventType) -> Self {
+        match value {
+            EventType::MessageSend => 1,
+            EventType::Unknown(unknown) => unknown,
+        }
+    }
+}
+
+/// Characterizes the type of content which can trigger the rule.
+#[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
+pub enum Trigger {
+    Keyword(Vec<String>),
+    HarmfulLink,
+    Spam,
+    KeywordPresent(Vec<KeywordPresentType>),
```

Comment:
A bunch of coordinates typoes: Present -> Preset

Reply:
oof :sweat:

Reply:
Haha don't feel bad it happens

Reply:
yeah, happens when writing/reading is replaced by autocomplete/pattern-matching.